pjmedia/src/pjmedia-audiodev/alsa_dev.c

```diff
@@ -610,7 +610,13 @@ static int ca_thread_func (void *arg)
 
     while (!stream->quit) {
         pjmedia_frame frame;
-
+        
+        snd_pcm_uframes_t avail;
+        avail = snd_pcm_avail(pcm);
+        if (avail > 0 && avail <= nframes) {
```

Comment:
shouldn't this be `>=0 and < nframes`? We shouldn't need to sleep and continue when we already have `nframes`.

Reply:
when snd_pcm_prepare called,the snd_pcm_state is SND_PCM_STATE_PREPARED,snd_pcm_avail will return 0 always,need call snd_pcm_readi or snd_pcm_stop to change state to SND_PCM_STATE_RUNNING. i will update code with snd_pcm_state check later.